packages/compass-connections-navigation/src/connections-navigation-tree.tsx

```diff
@@ -378,68 +390,52 @@ const ConnectionsNavigationTree: React.FunctionComponent<
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-expect-error ignoring test props so they are not part of the interface
   __TEST_REACT_WINDOW_OVERSCAN = null,
-  ...restProps
 }) => {
   // we'll have either connections for MC version, or databases for SC version
-  const {
-    isSingleConnection,
-    data,
-  }: {
-    isSingleConnection: boolean;
-    data:
-      | MCConnectionsNavigationTreeProps['connections']
-      | SCConnectionsNavigationTreeProps['databases'];
-  } = useMemo(() => {
-    if ((restProps as SCConnectionsNavigationTreeProps).databases) {
-      return {
-        isSingleConnection: true,
-        data: (restProps as SCConnectionsNavigationTreeProps).databases,
-      };
-    }
-    return {
-      isSingleConnection: false,
-      data: (restProps as MCConnectionsNavigationTreeProps).connections,
-    };
-  }, [restProps]);
+  const isSingleConnection = !usePreference(
+    'enableNewMultipleConnectionSystem'
+  );
 
   const listRef = useRef<List | null>(null);
   const id = useId();
 
   useEffect(() => {
     if (activeNamespace) {
-      onDatabaseExpand(activeNamespace, true);
+      for (const connection of connections) {
+        onDatabaseExpand(connection.connectionInfo.id, activeNamespace, true);
+      }
```

Comment:
we won't need to call this multiple times, or? perhaps we can keep it now to not break the legacy version. but I'd add a TODO for when we have active connection, then we can just `onDatabaseExpand(activeConnection, activeNamespace, true)`.
similar to the todo below, where I put the wrong ticket number 🙈 . It will rather be part of @himanshusinghs's workspace changes